virtcontainers/utils/utils_linux.go

```diff
@@ -22,7 +22,7 @@ const ioctlVhostVsockSetGuestCid = 0x4008AF60
 
 var ioctlFunc = ioctl
 
-var maxUInt uint32 = 1<<32 - 1
+var maxUInt uint64 = 1<<32 - 1
```

Comment:
Given its type, this *looks* like it should be `1<<64 - 1`, so it might be worth referencing [Virtual I/O Device (VIRTIO) Version 1.0, Committee Specification 04, 03 March 2016, section "5.7.4 Device configuration layout"](http://stefanha.github.io/virtio) which states that although the contextID is 64-bits, 

> The upper 32 bits of the CID are reserved and zeroed.

In fact, maybe we need a few more unit tests to ensure we don't generate CID's beyond the 32-bit range of this 64-bit type?

Reply:
@jodh-intel I can add the reference. In govmm, I add the check if the cid is in the rage of 32 bits ( see https://github.com/intel/govmm/blob/900f3a1f182557ffd6dd582746ab32005a57d5a6/qemu/qemu.go#L1047). There is also a test case (see https://github.com/intel/govmm/blob/900f3a1f182557ffd6dd582746ab32005a57d5a6/qemu/qemu_test.go#L315). I can add few test cases in this PR or open a new one.

Reply:
@jodh-intel I update the comment. I don't know how can I set a wrong cid in kata. You always generate the cid with `FindContextID`and there isn't a check if the context id is valid.

Reply:
Hi @alicefr - Agreed - that sounds like the best place to perform the checks.